bin/check-es-maximum-shard-limit.rb

```diff
@@ -0,0 +1,138 @@
+#! /usr/bin/env ruby
+#
+# Checks to see if there are too many shards in the cluster
+# ===
+#
+# DESCRIPTION:
+#   Checks to see if the number of shards in the Elasticsearch cluster
+#   breaches a limit based on the number of nodes in the cluster and
+#   a configurable limit per node
+#
+# OUTPUT:
+#   plain-text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: rest-client
+#
+# USAGE:
+#
+# NOTES:
+#   This check will check the limit across the cluster, it does not check the
+#   shard limit on a node by node basis
+#
+# LICENSE:
+#   Thomas Riley
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'rest-client'
+require 'json'
+require 'base64'
+
+#
+# == Elastic Search Shard Limit Status
+#
+class ESShardMaximumLimit < Sensu::Plugin::Check::CLI
+  option :scheme,
```

Comment:
I'd prefer to use `protocol` as `scheme` sounds more like the scheme in a metrics script. I would also accept `uri_protocol` or something like that to make it obvious to it's intent.

Also I would suggest including the only valid options like this:
```ruby
option :protocol,
           description: 'Protocol to make requests with',
           long: '--protocol PROTO',
           default: 'http',
           in: %(http https)
```